src/Api/EncodedString.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "include/Api/EncodedString.h"
+
+#include <absl/base/casts.h>
+
+#include "capture.pb.h"
+
+namespace orbit_api {
+
+namespace {
+inline void DecodeString(uint64_t encoded_name, std::string* out) {
```

Comment:
Hmmm, I think the entire class could be made simpler. In the end, we are using 8 uint64_t as a buffer to encode a string. I'd suggest using a union:

union EncodedString{
  uint64_t chunks[8];
  char str[64];
}

This way we can std::strncpy a string into the `str` bufffer and also access the individual uint64_t chunks. WDYT?

Also, if the fixed64 members in the proto are contiguous in memory, this could be done in-place.

Reply:
Similar to what's done here: 

https://github.com/google/orbit/blob/8cc1104be26a3da8589eb35f8d6904819e348957/src/Api/include/Api/EncodedEvent.h#L42

Reply:
Following @dpallotti's suggestion, I aligned the code with the usual endianess, allowing to call `strncpy`. That simplified the code a lot. However, I would highly prefer to do the 8 copies individually, rather than using the union. Also note, that the union would not cover the dynamically sized "addtional" part of the string.

Reply:
Also, protobuf does not give us direct access to the base type fields. So even if the proto fields would be continuous in memory, we could not take advantage of it.

Reply:
> note, that the union would not cover the dynamically sized "additional" part of the string
Yes of course. The suggestion was for the fixed part of the string. But yeah, fair enough.